### PR TITLE
ATM: Boost XssThroughDOM

### PR DESCRIPTION
**Pull request by tiferet**
This is a new version of https://github.com/github/codeql/pull/11204 based on top of the CodeQL refactoring, of which https://github.com/github/codeql/pull/11323 is the final PR.

@jhelie Because rebasing was too hard, I created this new version of [your PR](https://github.com/github/codeql/pull/11204) based on top of the new CodeQL framework. As I did this, though, I noticed that, as I understand it, there are more changes that need to be made when boosting a new query that are missing from your PR. Searching the repo for the substring `XssAtm` I found changes that I think need to be made for `XssThroughDOM` in the following files:
- javascript/ql/experimental/adaptivethreatmodeling/test/endpoint_large_scale/EndpointFeatures.ql
- javascript/ql/experimental/adaptivethreatmodeling/test/endpoint_large_scale/FilteredTruePositives.ql
- javascript/ql/experimental/adaptivethreatmodeling/modelbuilding/DebugResultInclusion.ql
- javascript/ql/experimental/adaptivethreatmodeling/modelbuilding/extraction/ExtractEndpointDataTraining.qll (this one will go away if we deprecate `hasFlowFromSource` from the training data)

The changes to the tests will also require changes to the `.expected` files, of course.

### CODE REVIEW

File: javascript/ql/experimental/adaptivethreatmodeling/src/XssThroughDomATM.ql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *       external/cwe/cwe-079 external/cwe/cwe-116
+ */
+
+import javascript
+import ATM::ResultsInfo
+import DataFlow::PathGraph
+import experimental.adaptivethreatmodeling.XssThroughDomATM
+
+from AtmConfig cfg, DataFlow::PathNode source, DataFlow::PathNode sink, float score
+where cfg.getAlerts(source, sink, score)
```
[the remark this answers is not in the document]
Comment (jhelie) [reply javascript/ql/experimental/adaptivethreatmodeling/src/XssThroughDomATM.ql:22]: @henrymercer any clue?

**Comment by jhelie** [comment]
I'm not sure why the tests are failing, I can run the query on my machine and the tests pass.

**Comment by henrymercer** [comment]
The tests run on a merge commit, and there have been changes on main since this PR.  So I think we can resolve this by merging in `main` and updating the code, e.g. the `getAlerts` predicate specifically.

**Comment by jhelie** [comment]
I've merged main but I'm not sure which piece of code I'm supposed to update?

**Comment by tiferet** [comment]
Note: When https://github.com/github/codeql/pull/11532 gets merged XssThroughDOM should be added to that test as well.

**Review by adityasharad — APPROVED** [review]
Code changes look good. Assuming no other concerns from the team, I recommend merging this as is, so we don't accumulate conflicts with other PRs, and evaluating the combined state of `main`.

File: javascript/ql/experimental/adaptivethreatmodeling/src/XssThroughDomATM.ql [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/**
+ * For internal use only.
```
[the remark this answers is not in the document]
Comment (jhelie) [reply javascript/ql/experimental/adaptivethreatmodeling/src/XssThroughDomATM.ql:2]: This comment is present in all ATM-related `.ql` and `.qll` so probably best done in a separate PR given the churn. I'll open one.

**Comment by jhelie** [comment]
Merging now, and will evaluate new `main` state.

**Pull request merged**